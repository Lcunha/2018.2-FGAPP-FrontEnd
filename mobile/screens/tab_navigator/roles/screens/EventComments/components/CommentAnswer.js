import {
    Button,
    Icon,
    Text,
    ActionSheet,
    Root
} from 'native-base';
import React, { Component } from 'react';
import { View } from 'react-native';
import * as helpers from '../../../utils/helpers';
// import LikeButton from './LikeButton';

const BUTTONS = [
    { text: 'Responder', icon: 'quote', iconColor: '#2c8ef4' },
    // { text: 'Editar', icon: 'hammer', iconColor: '#f42ced' },
    { text: 'Deletar', icon: 'trash', iconColor: '#fa213b' },
    { text: 'Cancelar', icon: 'close', iconColor: 'rgb(0,0,0)' }
];
const DESTRUCTIVE_INDEX = 1;
const CANCEL_INDEX = 2;

class CommentItem extends Component {
    state = {};

    _menuHandler = option => {
        switch (option) {
            case 0:
                // console.log('Apertou botão de resposta.');
                break;
            case 1:
                this.props.onDelete(this.props.id);
                break;

            case 2:
                break;
        }
    };

    render() {
        const { author, text, postDate } = this.props;
        return (
            <View
                style={{
                    marginLeft: '20%',
                    marginTop: 15,
                    borderLeftWidth: 1,
                    borderLeftColor: 'grey',
                    paddingLeft: 5
                }}
            >
                <View
                    style={{
                        flexDirection: 'row',
                        width: '100%',
                        justifyContent: 'space-between',
                        alignItems: 'center'
                    }}
                >
                    <Icon name="person" />
                    <Text
                        style={{
                            fontWeight: 'bold'
                        }}
                    >
                        {author}
                    </Text>
                    <Button
                        transparent
                        onPress={() =>
                            ActionSheet.show(
                                {
                                    options: BUTTONS,
                                    cancelButtonIndex: CANCEL_INDEX,
                                    destructiveButtonIndex: DESTRUCTIVE_INDEX,
                                    title: 'Comentário de ' + author
                                },
                                buttonIndex => this._menuHandler(buttonIndex)
                            )
                        }
                    >
                        <Icon name="more" />
                    </Button>
                </View>

                <Text>{text}</Text>
                <Text style={{ color: 'grey', fontSize: 12 }}>
                    Postado em: {helpers.datetimeParser(postDate)}
                </Text>
            </View>
        );
    }
}

export default CommentItem;
