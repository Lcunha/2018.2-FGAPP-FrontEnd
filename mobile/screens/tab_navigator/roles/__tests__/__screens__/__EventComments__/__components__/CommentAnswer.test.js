import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import CommentAnswer from '../../../../screens/EventComments/components/CommentAnswer'
Enzyme.configure({adapter: new Adapter()});

it('renders correctly', () => {
  const tree = renderer.create(<CommentAnswer />).toJSON();
  expect(tree).toMatchSnapshot();
});
