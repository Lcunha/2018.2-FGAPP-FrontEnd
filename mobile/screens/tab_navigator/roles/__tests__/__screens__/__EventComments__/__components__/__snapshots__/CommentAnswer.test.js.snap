// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "borderLeftColor": "grey",
      "borderLeftWidth": 1,
      "marginLeft": "20%",
      "marginTop": 15,
      "paddingLeft": 5,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "fontWeight": "bold",
          },
        ]
      }
      uppercase={false}
      virtual={undefined}
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 0,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#000",
        "fontFamily": "System",
        "fontSize": 16,
      }
    }
    uppercase={false}
    virtual={undefined}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        },
        Object {
          "color": "grey",
          "fontSize": 12,
        },
      ]
    }
    uppercase={false}
    virtual={undefined}
  >
    Postado em: 
    1
  </Text>
</View>
`;
